### DISCUSSION TITLE
[4.1] Add to Category List View

### DISCUSSION THREAD

**Discussion by rytechsites** [joomla/joomla-cms#18457] [2. Cold Features]
### Steps to reproduce the issue
Add the capability of adding Custom Fields to the Category List menu item (or tagged item list)

We should not have to create an 'html' override' in order to include the Joomla CORE Feature of custom fields to be displayed in the Category List format.

**Comment by tonypartridge** [comment]
Ok, so the problem you have is you can not anticipate what fields would be created and where they should be positioned. So saying they should be included means the core would have to do some serious magic which is just not possible.

So you need to create a template override to customise the display as you want it. There are components which you can use to develop the view and a plug for EasyLayouts lets you drag and drop to create you template override view.

**Comment by rytechsites** [comment]
Tony,
Thank you so much for your response and feedback...

So preferrably, I would LOVE the Joomla team to 'provide this magic'.

There is a tool, that Olivier Buissard created, Latest News Enhanced, which is a 'blog/component' and a module, he added an 'information tab', that allows you to add fields, he uses a 'dynamic drop down list', so it can include any of the custom fields that were created.

So ideally, in the Category LIST function, you would be able to add an additional tab (or include in the 'field list' screen), some kind of 'drop down', so you can dynamically select which field(s) you want displayed.

A simple 'override', would just be 'hard coding' the field you want in a specific view, which I know could work...but we manage over 120 Joomla sites and as you know the fields could be different, and we do not want to have overrides with different fields, view, etc...

Which is why I am asking for this 'feature request' to be added to some 'list' somewhere.

It was so great when Joomla finally added the custom fields, and I know those can be listed in a 'blog view', and I can include them in Latest News...but what our clients are really looking for is a 'list / responsive table' type view.  This is where the magic of your team comes in...maybe google summer of code?

I really appreciate your time and thought in this.  If you would like me to put this somewhere else, please let me know, I just want my ideas and thoughts to be heard in the right location (feature requests, etc).

I remember years ago, when custom fields was just a 'feature request', and Tags were just a 'feature request'...and over time they did come about in core.

thanks,
Laura

**Comment by pepperstreet** [comment]
CC: @laoneo  

Hello, I think there is no "magic" needed. ;) 
I agree with @rytechsites that every view should provide a default output for the fields. If there is a general field option for automatic display, it should display the fields somewhere... otherwise users are confused and wonder why no fields show-up at all. BTW, the current blog view follows the same logic. 

I am aware that the list/table view has its peculiarities and certain use cases. 
However it might display the fields like so: 
- right after the title: below it, in the same cell;
- before/after content display could be a new row as well; 
- groups might be a separate column;
- etc. 

---

_(Personal Note: Create a separate issue for…)_  
**A general and even more crucial issue** with the current default output: 
Automatic display applies to full-view **and** list-views :( 
That's really a drawback and a missing core feature. It should be possible to choose the view-context, like we can choose the site/admin/both display. Or could it be a menuItem parameter, or a component option.

**Comment by rytechsites** [comment]
Is there any update on this feature?

As our sites use fields more and more, it seems like being able to select fields to display in the category list is even more important for core.

thank you!

**Comment by ghost** [comment]
@rytechsites it needs a Dev. having Interest and Time.

**Comment by rytechsites** [comment]
Should this be marked to a [4.0] request since it would be a FEATURE?  (A very important feature), so it doesn't 'get lost' in joomla 3.x?

**Comment by ghost** [comment]
@rytechsites you're right, please add it at Start of Title.

**Comment by ghost** [comment]
@infograf768 please change Label to "J4 Issue".

**Comment by rytechsites** [comment]
Can this be reviewed for Joomla 4.1?

**Reply by tonypartridge** [reply]
Can you propose a desired approach to how it should work from a UX perspective?

**Reply by rytechsites** [reply]
Ability to add a single or multiple field, with a dropdown select (multi select):
![list-layout](https://user-images.githubusercontent.com/17861472/148622582-3fa64328-f34e-4d99-b30b-d4cfe1cfb5d0.jpg)

Result will display the field(s) within the category list as additional columns:
![result-layout](https://user-images.githubusercontent.com/17861472/148622576-3bed4310-eff6-4eb9-bd21-890365d9be4c.jpg)

**Comment by roland-d** [comment]
I do think this is a good addition to Joomla.

**Comment by tonypartridge** [comment]
Yes I agree, we have a similar method with JEvents List of Events. It makes
responsiveness much harder.

I’ll try and come up with a solution this month.

On Sat, 8 Jan 2022 at 19:38, Roland Dalmulder ***@***.***>
wrote:

> I do think this is a good addition to Joomla.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/discussions/18457#discussioncomment-1932909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKWBFS2MX35NQ7CO4JD3J3UVCHDHANCNFSM5LNI4BCA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Tony Partridge
Xtech Web Services Ltd

Managing Director/Developer

**Comment by brianteeman** [comment]
Take a look at this plugin - maybe it will help

https://alterbrains.com/joomla-extensions/content/alter-fields
